From ec8c74119e77460781dbf34188e425987002c12b Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Mon, 25 Jan 2016 15:18:51 +0200
Subject: [PATCH 0725/1240] clock: a3700: Add new clock preset - 800/800

- Add preset for CPU clock 800MHz and DDR clock 800MHz

Change-Id: Icc6c110636762e20b021d4aa87d268ac15c47057
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/27002
Reviewed-by: Omri Itach <omrii@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Tested-by: Hanna Hawa <hannah@marvell.com>
---
 arch/arm/include/asm/arch-armadalp/clock.h     | 63 ++++++++++++++++++++++++++
 doc/device-tree-bindings/clock/mvebu-clock.txt |  1 +
 include/dt-bindings/clock/freq_data.h          |  1 +
 3 files changed, 65 insertions(+)

diff --git a/arch/arm/include/asm/arch-armadalp/clock.h b/arch/arm/include/asm/arch-armadalp/clock.h
index 71e17c9..03ead95 100644
--- a/arch/arm/include/asm/arch-armadalp/clock.h
+++ b/arch/arm/include/asm/arch-armadalp/clock.h
@@ -415,6 +415,53 @@ TBG-B_S	1000	5	1	sec_at_pclk_sel		200	NB	DIV1[5:3]	DIV1[2:0]
 -------------------------------------------------------------------------------------------------------------------------------------
 TBG-B-P	500	2	5	gbe_50_pclk_sel		50	SB					DIV2[11:9]	DIV2[8:6]
 *************************************************************************************************************************************
+*************************************************************************************************************************************
+Configuration 3 – CPU 800, DDR 800
+
+		SE DIV	DIFF DIV
+KVCO-A	1600	0	1	← 2^n
+KVCO-B	2000	1	2	← 2^n						NB				SB
+=====================================================================================================================================
+		PRE-1	PRE-2	TARGET			FREQ	BR	PRE-1		PRE-2		PRE-1		PRE-2
+-------------------------------------------------------------------------------------------------------------------------------------
+TBG-A-S	1600	4	1	mmc_pclk_sel		400	NB	DIV2[15:13]	DIV2[18:16]
+	2^n →	1	NA	ddr_phy_mck_div_sel	800	NB	DIV0[18]	NA
+		4	1	sdio_pclk_sel		400	SB					DIV0[8:6]	DIV0[5:3]
+-------------------------------------------------------------------------------------------------------------------------------------
+		PRE-1	PRE-2	TARGET			FREQ	BR	PRE-1		PRE-2		PRE-1		PRE-2
+-------------------------------------------------------------------------------------------------------------------------------------
+TBG-A-P	800	5	NA	counter_pclk_sel	160	NB	DIV0[25:23]	NA
+	2^n →	1	NA	cpu_cnt_clk_div_sel	80	NB	DIV2[28:27]	NA
+		4	1	sqf_pclk_sel		200	NB	DIV1[26:24]	DIV1[29:27]
+		4	1	sata_host_pclk_sel	200	NB	DIV2[9:7]	DIV2[12:10]
+		4	1	tsecm_pclk_sel		200	NB	DIV1[14:12]	DIV1[17:15]
+		2	1	eip97_pclk_sel		400	NB	DIV2[24:22]	DIV2[21:19]
+		1	NA	setm_tmx_pclk_sel	800	NB	DIV1[20:18]	NA
+		1	NA	trace_pclk_sel		800	NB	DIV0[22:20]	NA
+		1	NA	a53_cpu_pclk_sel	800	NB	DIV0[30:28]	NA
+		4	NA	wc_ahb_div_sel		200	NB	DIV2[2:0]	NA
+	2^n →	0	NA	atb_clk_div_sel		800	NB	DIV2[26:5]	NA
+	2^n →	1	NA	plkdbg_clk_div_sel	400	NB	DIV2[30:29]	NA
+-------------------------------------------------------------------------------------------------------------------------------------
+		PRE-1	PRE-2	TARGET			FREQ	BR	PRE-1		PRE-2		PRE-1		PRE-2
+-------------------------------------------------------------------------------------------------------------------------------------
+TBG-B_S	1000	5	1	sec_at_pclk_sel		200	NB	DIV1[5:3]	DIV1[2:0]
+		4	2	ddr_fclk_pclk_sel	125	NB	DIV0[17:15]	DIV0[14:12]
+		5	2	sec_dap_pclk_sel	100	NB	DIV1[11:9]	DIV1[8:6]
+		5	2	pwm_pclk_sel		100	NB	DIV0[5:3]	DIV0[2:0]
+		4	1	gbe_core_pclk_sel	250	SB					DIV1[20:18]	DIV1[23:21]
+		1	NA	gbe_bm_core_clk_div2_sel250	SB					DIV1[12]	NA
+		1	NA	gbe1_core_clk_div2_sel	250	SB					DIV1[13]	NA
+		1	NA	gbe0_core_clk_div2_sel	250	SB					DIV1[14]	NA
+		4	2	gbe_125_pclk_sel	125	SB					DIV1[11:9]	DIV1[8:6]
+		2	2	sb_axi_pclk_sel		250	SB					DIV0[20:18]	DIV0[23:21]
+		4	2	usb32_ss_sys_pclk_sel	125	SB					DIV0[17:15]	DIV0[20:18]
+		5	2	usb32_usb2_sys_pclk_sel	100	SB					DIV0[14:12]	DIV0[11:9]
+-------------------------------------------------------------------------------------------------------------------------------------
+		PRE-1	PRE-2	TARGET			FREQ	BR	PRE-1		PRE-2		PRE-1		PRE-2
+-------------------------------------------------------------------------------------------------------------------------------------
+TBG-B-P	500	2	5	gbe_50_pclk_sel		50	SB					DIV2[11:9]	DIV2[8:6]
+*************************************************************************************************************************************
 */
 /* Init values for the static clock configurations array */
 #define MVEBU_A3700_CLOCK_CFGS	{\
@@ -466,6 +513,22 @@ TBG-B-P	500	2	5	gbe_50_pclk_sel		50	SB					DIV2[11:9]	DIV2[8:6]
 		{2, 4, 1, 1, 1, 4, 1},		/* DIV1 */\
 		{5, 2} }			/* DIV2 */\
 	},\
+	/* Clock tree configuration entry 3 */\
+	{ 800, 800, /* CPU, DDR */\
+		{1600, 0, 1}, /* TBG-A: KVCO Freq, SE_VCODIV, DIFF_VCODIV */\
+		{2000, 1, 2}, /* TBG-B: KVCO Freq, SE_VCODIV, DIFF_VCODIV */\
+		/* NorthBridge */\
+		{{TBG_A_S, TBG_A_P, TBG_B_S, TBG_B_S, TBG_A_P, TBG_A_P, TBG_A_P,\
+		  TBG_B_S, TBG_B_S, TBG_A_P, TBG_A_P, TBG_A_P, TBG_A_P},\
+		{4, 5, 2, 4, 1, 1, 5, 1},		/* DIV0 */\
+		{1, 5, 2, 5, 1, 4, 1, 1, 4},		/* DIV1 */\
+		{2, 1, 4, 1, 4, 2, 1, 0, 1, 1} },	/* DIV2 */\
+		/* SouthBridge */\
+		{{TBG_B_P, TBG_B_S, TBG_B_S, TBG_A_S, TBG_B_S, TBG_B_S, TBG_B_S},\
+		{1, 4, 2, 5, 2, 4, 2, 2},	/* DIV0 */\
+		{2, 4, 1, 1, 1, 4, 1},		/* DIV1 */\
+		{5, 2} }			/* DIV2 */\
+	},\
 }
 
 /*****************************************************************************
diff --git a/doc/device-tree-bindings/clock/mvebu-clock.txt b/doc/device-tree-bindings/clock/mvebu-clock.txt
index 3dda6cf..f2c1144 100644
--- a/doc/device-tree-bindings/clock/mvebu-clock.txt
+++ b/doc/device-tree-bindings/clock/mvebu-clock.txt
@@ -18,6 +18,7 @@ freq: define the clock preset for CPU/DDR
 		PRESET_CPU_600_DDR_600  - CPU 600MHz,  DDR 600MHz
 		PRESET_CPU_400_DDR_600  - CPU 400MHz,  DDR 600MHz
 		PRESET_CPU_1000_DDR_800 - CPU 1000MHz, DDR 800MHz
+		PRESET_CPU_800_DDR_800  - CPU 800MHz,  DDR 800MHz
 
 Armada 3700 example:
 soc {
diff --git a/include/dt-bindings/clock/freq_data.h b/include/dt-bindings/clock/freq_data.h
index 329bcfa..2971f66 100644
--- a/include/dt-bindings/clock/freq_data.h
+++ b/include/dt-bindings/clock/freq_data.h
@@ -22,5 +22,6 @@
 #define PRESET_CPU_600_DDR_600		0
 #define PRESET_CPU_400_DDR_600		1
 #define PRESET_CPU_1000_DDR_800		2
+#define PRESET_CPU_800_DDR_800		3
 
 #endif /* _FREQ_DATA_H_ */
-- 
1.9.1

